packages/x-data-grid/src/hooks/features/rowSelection/utils.ts

```diff
@@ -1,5 +1,110 @@
-import type { DataGridProcessedProps } from '../../../models/props/DataGridProps';
 import { GridSignature } from '../../utils/useGridApiEventHandler';
+import { GRID_ROOT_GROUP_ID } from '../rows/gridRowsUtils';
+import { gridFilteredRowsLookupSelector } from '../filter/gridFilterSelector';
+import { gridSortedRowIdsSelector } from '../sorting/gridSortingSelector';
+import { selectedIdsLookupSelector } from './gridRowSelectionSelector';
+import { gridRowTreeSelector } from '../rows/gridRowsSelector';
+import { createSelector } from '../../../utils/createSelector';
+import type { GridGroupNode, GridRowId, GridRowTreeConfig } from '../../../models/gridRows';
+import type { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import type {
+  GridPrivateApiCommunity,
+  GridApiCommunity,
+} from '../../../models/api/gridApiCommunity';
+import type { GridRowSelectionPropagation } from '../../../models/gridRowSelectionModel';
+
+export const ROW_SELECTION_PROPAGATION_DEFAULT: GridRowSelectionPropagation = {
+  parents: false,
+  descendants: false,
+};
+
+const isValidRowId = (id: GridRowId | null): id is GridRowId =>
+  id !== null && (typeof id === 'number' || typeof id === 'string');
```

Comment:
Wouldn't `id != null` be enough here?

Reply:
True. `0 == null` evaluating to `false` is confusing for me as both are falsy values and many times I get deceived by it, thus I tried to add the extra check here as per `GridRowId` type.

Technically `id !== null` could also mean some complex value like an object will also pass the check (example: `{ a: 1 } != null`). But in this case, simplifying makes sense since it's not likely to be passed in this function.

Reply:
Since `type GridRowId = number | string`
Would `typeof id === 'number' || typeof id === 'string'` be even better as it will ensure runtime type safety?

Reply:
That's interesting, for me `id != null` is super readable and I auto translate it to "it can't be nullable" (so `null` or `undefined`)

But I dislike `!!id` strongly because it's super easy to forget some falsy values (`0` and `""` most of the time).

Reply:
Yeah, I guess readability is a relative term sometimes, it could be different for two people. For my case, what looks weird to me around `id == null` is it's non-conventional behavior with value `0`
```tsx
console.log(0 == null) // `false` (even though both are falsy values)
console.log(!0 == !null) // `true`
console.log(Boolean(0) == Boolean(null)) // `true`
```

Reply:
```ts
console.log(0 == null) // `false` (even though both are falsy values)
```

But `==` does not check if values are falsy or truthy, you wouldn't expect `console.log(1 == 2)` to return true because both are truthy.

As you said, readability is super relative.
I think I used `==` so little outside of `==` in the past few years that for me `== null` probably became a standalone operator, something like `isNullable()`.

Reply:
Cool. 👍 Thanks for your input anyway.
I'll consider this one resolved since `id != null` seems to work in this particular use case.

Feel free to reopen in case of a disagreement.

For a separate topic though, I'd like to have a recommended way of doing `isValidRowId` that works in all the cases.